pkg/util/prometheushelper/helper.go

```diff
@@ -0,0 +1,168 @@
+
+var (
+	validMetric = regexp.MustCompile(`^[a-zA-Z_:][a-zA-Z0-9_:]*$`)
+	validLabel  = regexp.MustCompile(`^[a-zA-Z_][a-zA-Z0-9_]*$`)
+)
+
+// NewCounter create the counter metric
+func NewCounter(metric string, help string, labels []string) *prometheus.
+	CounterVec {
+
+	metricName, err := getAndValid(metric, labels)
+	if err != nil {
+		logger.Errorf("[%s] %v", module, err)
+		return nil
+	}
+
+	if m, find := counterMap[metricName]; find {
+		logger.Debugf("[%s] Counter <%s> already created!", module, metricName)
+		return m
+	}
+
+	counterMap[metricName] = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: metricName,
+			Help: help,
+		},
+		labels,
+	)
+	prometheus.MustRegister(counterMap[metricName])
+
+	logger.Infof("[%s] Counter <%s> is created!", module, metricName)
+	return counterMap[metricName]
+}
+
+// NewGauge create the gauge metric
+func NewGauge(metric string, help string, labels []string) *prometheus.
+	GaugeVec {
+
+	metricName, err := getAndValid(metric, labels)
+	if err != nil {
+		logger.Errorf("[%s] %v", module, err)
+		return nil
+	}
+	if m, find := gaugeMap[metricName]; find {
+		logger.Debugf("[%s] Gauge <%s> already created!", module, metricName)
+		return m
+	}
+
+	gaugeMap[metricName] = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: metricName,
+			Help: help,
+		},
+		labels,
+	)
+	prometheus.MustRegister(gaugeMap[metricName])
+
+	logger.Infof("[%s] Gauge <%s> is created!", module, metricName)
+	return gaugeMap[metricName]
+}
+
+// NewHistogram create the Histogram metric
+func NewHistogram(metric string, help string, labels []string) *prometheus.HistogramVec {
+
+	metricName, err := getAndValid(metric, labels)
+	if err != nil {
+		logger.Errorf("[%s] %v", module, err)
+		return nil
+	}
+	if m, find := histogramMap[metricName]; find {
+		logger.Debugf("[%s] Histogram <%s> already created!", module, metricName)
+		return m
+	}
+
+	histogramMap[metricName] = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name: metricName,
+			Help: help,
+		},
+		labels,
+	)
+	prometheus.MustRegister(histogramMap[metricName])
+	logger.Infof("[%s] Histogram <%s> already created!", module, metricName)
+	return histogramMap[metricName]
+}
+
+// NewSummary create the NewSummary metric
+func NewSummary(metric string, help string, labels []string) *prometheus.SummaryVec {
+
+	metricName, err := getAndValid(metric, labels)
+	if err != nil {
+		logger.Errorf("[%s] %v", module, err)
+		return nil
+	}
+	if m, find := summaryMap[metricName]; find {
+		logger.Debugf("[%s] Summary <%s> already created!", module, metricName)
+		return m
+	}
+
+	summaryMap[metricName] = prometheus.NewSummaryVec(
+		prometheus.SummaryOpts{
+			Name: metricName,
+			Help: help,
+		},
+		labels,
+	)
+	prometheus.MustRegister(summaryMap[metricName])
+	logger.Infof("[%s] Summary <%s> already created!", module, metricName)
+	return summaryMap[metricName]
+}
+
+func getAndValid(metricName string, labels []string) (string, error) {
```

Comment:
`valid` is an adjective.

```suggestion
func getAndValidate(metricName string, labels []string) (string, error) {
```